### DISCUSSION TITLE
想问下为什么最新的skynet在snax服务里require "snax"会报错呢

### DISCUSSION THREAD

**Discussion by Luoweihong** [cloudwu/skynet#1676] [General]
在main里启动了一个snaxTest 的snax服务，然后在snaxTest的开头local snax = require "snax"
在以前的老服务器里边这样子用没问题。。新手犯懵了
[:0000000d] LAUNCH snlua snaxd snaxTest
[:0000000d] lua loader error : /home/skynet_new/skynet/lualib/snax/interface.lua:82: /home/skynet_new/skynet/lualib/skynet.lua:52: assertion failed!
stack traceback:
	[C]: in function 'assert'
	/home/skynet_new/skynet/lualib/skynet.lua:52: in function 'skynet.register_protocol'
	/home/skynet_new/skynet/lualib/skynet/snax.lua:11: in main chunk
	[C]: in upvalue 'require'
	/home/skynet_new/skynet/lualib/skynet/require.lua:31: in function 'skynet.require.require'
	/home/houseTimePopmart/snax/snaxTest.lua:5: in main chunk
	[C]: in function 'xpcall'
	/home/skynet_new/skynet/lualib/snax/interface.lua:80: in function 'snax.interface'
	/home/skynet_new/skynet/service/snaxd.lua:10: in local 'main'
	/home/skynet_new/skynet/lualib/loader.lua:50: in main chunk
stack traceback:
	[C]: in function 'assert'
	/home/skynet_new/skynet/lualib/snax/interface.lua:82: in function 'snax.interface'
	/home/skynet_new/skynet/service/snaxd.lua:10: in local 'main'
	/home/skynet_new/skynet/lualib/loader.lua:50: in main chunk
[:0000000d] KILL self
[:00000008] init service failed: /home/skynet_new/skynet/lualib/skynet.lua:590: call failed
stack traceback:
	[C]: in function 'error'
	/home/skynet_new/skynet/lualib/skynet.lua:590: in upvalue 'yield_call'
	/home/skynet_new/skynet/lualib/skynet.lua:607: in function 'skynet.call'
	(...tail calls...)
	/home/skynet_new/skynet/lualib/skynet/snax.lua:86: in function 'snax.rawnewservice'
	/home/skynet_new/skynet/lualib/skynet/snax.lua:107: in function 'snax.newservice'
	/home/houseTimePopmart/main.lua:35: in upvalue 'start'
	/home/skynet_new/skynet/lualib/skynet.lua:935: in function </home/skynet_new/skynet/lualib/skynet.lua:933>
	[C]: in function 'xpcall'
	/home/skynet_new/skynet/lualib/skynet.lua:937: in function 'skynet.init_service'
	/home/skynet_new/skynet/lualib/skynet.lua:950: in upvalue 'f'
	/home/skynet_new/skynet/lualib/skynet.lua:253: in function </home/skynet_new/skynet/lualib/skynet.lua:252>

**Comment by cloudwu** [comment]
require “skynet.snax”
